### DISCUSSION TITLE
Delay in output

### DISCUSSION THREAD

**Discussion by Aadi4401** [ultralytics/ultralytics#14502] [Q&A]
Hello,
I am using a CPU for the object detection process and utilizing 3 to 4 .pt files. When I run YOLOv8 code with video, webcam, or RTSP feed, the output video has a delay of 5 to 6 seconds. Is there any way to get the output without delay? If the delay issue is due to using multiple .pt files, then is there a way to merge these files into one .pt file and use it?

**Comment by pderrenger** [comment]
@Aadi4401 hi there! The delay you're experiencing is likely due to the computational load on the CPU, especially when using multiple .pt files. Merging models isn't straightforward, but you can try optimizing your setup by using a more powerful GPU if available or reducing the model size. Additionally, ensure you're using the latest version of YOLOv8, as performance improvements are continually being made. If the issue persists, consider simplifying the models or the input data to reduce processing time.
